trin-core/src/portalnet/storage.rs

```diff
@@ -141,6 +148,24 @@ impl PortalStorage {
         &mut self,
         key: &impl OverlayContentKey,
         value: &Vec<u8>,
+    ) -> Result<(), PortalStorageError> {
+        match self.durable_store(key, value) {
+            Ok(()) => Ok(()),
+            Err(e) => {
+                // When the data cannot be stored locally, save it to an LRU. This supports
+                // a few use cases, like:
+                // - offering all portal_historyStore content to peers, ignoring radius
+                // - serving hotspots in network, outside of radius
+                self.cache.write().put(key.content_id(), value.clone());
+                Err(e)
```

Comment:
I think returning an error here is a little dangerous because the caller will not expect this method to change any external state (in this case LRU) when this method returns an error.

Reply:
Yeah, I'm considering reverting this whole PR anyway. If it was going to work without any fiddling I might have been happy, but there are just too many little quirks like this one. Another one I've seen is some kind of surprising race condition or something that causes the data to be missing, occasionally.

I think I see the path to the alternative now. To thread the values through in memory, so that we don't rely on the database to answer the Accept.